File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> sizes) 
{
    int wMax = 0;
    int hMax = 0;
    for(int i=0; i<sizes.size(); i++)
    {
        int w = sizes[i][0];
        int h = sizes[i][1];

        if(wMax < w)
        {
            wMax = w;
        }
        if(hMax < h)
        {
            hMax = h;
        }
    }

    int max = (wMax > hMax) ? wMax : hMax;

    vector<int> ans(2,0);
    ans[0] = max;

    for(int i=0; i<sizes.size(); i++)
    {
        int w = sizes[i][0];
        int h = sizes[i][1];

        int tmp = (w > h) ? h : w;
        if(ans[1] < tmp)
        {             
            ans[1] = tmp;
        }
    }
    int area = 1;
    for(int &num : ans)
    {
        area *= num;
    }

    return area;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cs
using System;

public class Solution 
{
    public int solution(int[,] sizes) 
    {
        int[] wallet = new int[2];
        
        // 가로가길고 세로가짧게 정렬
        for(int i=0; i<sizes.GetLength(0); i++)
        {
            if(sizes[i,0] < sizes[i,1])
            {
                int tmp = sizes[i,0];
                sizes[i,0] = sizes[i,1];
                sizes[i,1] = tmp;
            } 
        }
        wallet[0] = sizes[0,0];
        wallet[1] = sizes[0,1];
        
        for(int i=0; i<sizes.GetLength(0); i++)
        {
            if(wallet[0] < sizes[i,0])
            {
                wallet[0] = sizes[i,0];
            }
            
            if(wallet[1] < sizes[i,1])
            {
                wallet[1] = sizes[i,1];
            }
        }
        Console.WriteLine($"{wallet[0]}, {wallet[1]}");
        return wallet[0] * wallet[1];
    }
}